pkg/services/sqlstore/endpoint.go

```diff
@@ -800,6 +800,45 @@ func batchUpdateCheckState(sess *session, jobs []*m.AlertingJob) ([]*m.AlertingJ
 	return jobsWithStateChange, nil
 }
 
+func UpdateCheckState(job *m.AlertingJob) (bool, error) {
+	sess, err := newSession(true, "check")
+	stateChange := false
+	if err != nil {
+		return stateChange, err
+	}
+	defer sess.Cleanup()
+	stateChange, err = updateCheckState(sess, job)
+	if err != nil {
+		return stateChange, err
+	}
+	sess.Complete()
+	return stateChange, nil
+}
+
+func updateCheckState(sess *session, j *m.AlertingJob) (bool, error) {
+	stateSql := "UPDATE `check` SET state=?, state_change=? WHERE id=? AND state != ? AND state_change < ?"
+	lastCheckSql := "UPDATE `check` SET state_check=? WHERE id=? and state_check < ?"
+	stateChange := false
+
+	res, err := sess.Exec(stateSql, int(j.NewState), j.TimeExec, j.Id, int(j.NewState), j.TimeExec)
+	if err != nil {
+		return stateChange, err
+	}
+
+	aff, _ := res.RowsAffected()
+	if aff > 0 {
+		// state change.
+		stateChange = true
+	}
+
+	res, err = sess.Exec(lastCheckSql, j.TimeExec, j.Id, j.TimeExec)
+	if err != nil {
```

Comment:
this could just directly `return stateChange, err` instead of the `if` block and otherwise `return stateChange, nil`